File: ReshapeTheMatrix/ReshTheMat.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        m, n = len(mat), len(mat[0])
        if r * c != m * n:
            return mat
        res = [[0] * c for k in range(r)]
        for i in range(m * n):
            res[i // c][i % c] = mat[i // n][i % n]
        return res

File: ReshapeTheMatrix/ReshTheMat.js
/**
 * @param {number[][]} mat
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function (mat, r, c) {
  let arr = mat.flat();
  if (r * c != arr.length) return mat;
  let res = [];
  while (arr.length) res.push(arr.splice(0, c));
  return res;
};
